datasets/gb/proscribed_orgs/crawler.py

```diff
@@ -0,0 +1,145 @@
+INCLUDING1_RE = re.compile(r"(.*)\s*\(including\s+(.*)\)", re.IGNORECASE)
+INCLUDING2_RE = re.compile(r"(.*),?\s*including\s+(.*)", re.IGNORECASE)
+COMMA_AND_RE = re.compile(r",\s+|\s+and\s+", re.IGNORECASE)
+ALIAS_RE = re.compile(r"\s+\(([^)]+)\)")
+JUNK = " ,.;\t\r\n"
+DATE_FORMATS = [
+    "%d.%m.%Y"
+]
+
+
+def parse_comment(context: Context, text: str) -> List[str]:
+    """
+    Parse stack of  commentary from the act looking for the most
+    recent date when groups were added.
+    """
+    m = ADDITION_RE.search(text)
+    if m is None:
+        return []
+    return h.parse_date(m.group(2), DATE_FORMATS)
+
+
+def crawl_group(context: Context, text: str, change_stack: List[str]):
+    """
+    Process a group in the list along with associated stack of commentary.
+    """
+    text = text.strip(JUNK)
+    if not text:
+        return
+    add_date = []
+    for item in change_stack:
+        add_date = parse_comment(context, item)
+        if add_date:
+            break
+    context.log.info(f"Adding group: {text} (starting {add_date})")
+    entity = context.make("Organization")
+    entity.id = context.make_id(text)
+    entity.add("topics", "crime.terror")
+    context.log.debug(f"Unique ID {entity.id}")
+    names = []
+    m = INCLUDING1_RE.match(text)
+    if m:
+        names.append(m.group(1))
+        names.extend(COMMA_AND_RE.split(m.group(2)))
+        context.log.debug(f"Including(1): {names}")
+    else:
+        m = INCLUDING2_RE.match(text)
+        if m:
+            names.append(m.group(1))
+            names.extend(COMMA_AND_RE.split(m.group(2)))
+            context.log.debug(f"Including(2): {names}")
+        else:
+            names.append(text)
+    context.log.debug(f"Names and aliases: {names}")
+    alt_names = []
+    for name in names:
+        aka = []
+        for m in ALIAS_RE.finditer(name):
+            alias = m.group(1)
+            context.log.debug(f"alias: {alias}")
+            # Special case for single nested parenthesis
+            before, _, after = alias.partition("(")
+            if after:
+                aka.append(before.strip(JUNK))
+                aka.append(after.strip(JUNK))
+            else:
+                aka.append(alias.strip(JUNK))
+        aka.insert(0, ALIAS_RE.sub("", name).strip(JUNK))
+        alt_names.append(aka)
+    context.log.debug(f"Names and aliases: {alt_names}")
+    for aka in alt_names:
+        context.log.info(f"Adding primary name: {aka[0]}")
+        h.apply_name(entity, aka[0])
+        for alias in aka[1:]:
+            context.log.info(f"Adding alias: {alias}")
+            h.apply_name(entity, alias, alias=True)
+    sanction = h.make_sanction(context, entity)
+    if add_date:
+        sanction.add("startDate", add_date)
+    # For some reason make_sanction uses the top-level URL and
+    # publisher.name, which are not exactly accurate.
+    sanction.add("authority", "UK Home Secretary")
+    sanction.add("sourceUrl", context.data_url)
```

Comment:
no need to add these - make_sanction already does.

Reply:
Hi! There's a comment in there about this actually - the issue is that it takes the publisher as the authority, and the top-level URL as the source, which in this case is wrong, I think - the publisher is `legislation.co.uk` but the *authority* would be the Home Secretary.  Likewise, the descriptive top-level URL (the policy paper) is not the source URL for the data.

But I might not have written the metadata right - what would the best way to do this be?

Reply:
That's awkward - it's right there :) Sorry I missed the comment.

So Entity.add will add these values and leave the incorrect ones. I think these two URLs for this is quite handy actually, but I see there's an `Entity.set` you could use for `authority` - it's also used in eu_sanctions_map in this way.

Reply:
```suggestion
    sanction.set("authority", "UK Home Secretary")
    sanction.add("sourceUrl", context.data_url)
```